internal/scheduler/reports.go

```diff
@@ -89,102 +89,107 @@ func (report *SchedulingRoundReport) String() string {
 	fmt.Fprintf(w, "Finished:\t%s\n", report.Finished)
 	fmt.Fprintf(w, "Duration:\t%s\n", report.Finished.Sub(report.Started))
 	fmt.Fprintf(w, "Total capacity:\t%s\n", report.TotalResources.CompactString())
-	if len(report.PriorityFactorByQueue) == 0 {
-		fmt.Fprint(w, "Queue priority factors:\tnone\n")
-	} else {
-		fmt.Fprint(w, "Queue priority factors:\n")
-		for queue, priorityFactor := range report.PriorityFactorByQueue {
-			fmt.Fprintf(w, "  %s: %f\n", queue, priorityFactor)
-		}
-	}
 	totalJobsScheduled := 0
 	totalResourcesScheduled := make(schedulerobjects.QuantityByPriorityAndResourceType)
-	if len(report.SuccessfulJobSchedulingReportsByQueue) == 0 {
-		fmt.Fprint(w, "Jobs scheduled:\tnone\n")
-	} else {
-		fmt.Fprint(w, "Jobs scheduled:\n")
-		for queue, jobReports := range report.SuccessfulJobSchedulingReportsByQueue {
-			scheduledResourcesByPriority, ok := report.ScheduledResourcesByQueueAndPriority[queue]
-			if ok {
-				fmt.Fprintf(w, "  %s: %d (%s)\n", queue, len(jobReports), scheduledResourcesByPriority)
-			} else {
-				fmt.Fprintf(w, "  %s: %d\n", queue, len(jobReports))
-			}
-			totalJobsScheduled += len(jobReports)
-			totalResourcesScheduled.Add(scheduledResourcesByPriority)
-		}
-	}
 	fmt.Fprintf(w, "Total jobs scheduled:\t%d\n", totalJobsScheduled)
 	fmt.Fprintf(w, "Total resources scheduled:\t%s\n", totalResourcesScheduled)
 	fmt.Fprintf(w, "Termination reason:\t%s\n", report.TerminationReason)
 	w.Flush()
 	return sb.String()
 }
 
-// Add a job scheduling report to the report for this invocation of the scheduler.
+// AddJobSchedulingReport adds a job scheduling report to the report for this invocation of the scheduler.
 // Automatically updates scheduled resources by calling AddScheduledResources. Is thread-safe.
-func (schedulingRoundReport *SchedulingRoundReport) AddJobSchedulingReport(jobSchedulingReport *JobSchedulingReport) {
-	schedulingRoundReport.mu.Lock()
-	queue := jobSchedulingReport.Job.Queue
-	if jobSchedulingReport.UnschedulableReason == "" {
-		if m, ok := schedulingRoundReport.SuccessfulJobSchedulingReportsByQueue[queue]; ok {
-			m[jobSchedulingReport.JobId] = jobSchedulingReport
-		} else {
-			schedulingRoundReport.SuccessfulJobSchedulingReportsByQueue[queue] = map[uuid.UUID]*JobSchedulingReport{
-				jobSchedulingReport.JobId: jobSchedulingReport,
-			}
-		}
-	} else {
-		if m, ok := schedulingRoundReport.UnsuccessfulJobSchedulingReportsByQueue[queue]; ok {
-			m[jobSchedulingReport.JobId] = jobSchedulingReport
-		} else {
-			schedulingRoundReport.UnsuccessfulJobSchedulingReportsByQueue[queue] = map[uuid.UUID]*JobSchedulingReport{
-				jobSchedulingReport.JobId: jobSchedulingReport,
-			}
-		}
+func (report *SchedulingRoundReport) AddJobSchedulingReport(r *JobSchedulingReport) {
+	if report == nil {
```

Comment:
Why is it ok for this function to be null safe but others not?  In general I'm against making public functions null safe other than  in some special case (appending to slic es being a good example), because it risks an unintended null going unrecognised and then  surfacing as a problem in  some other random part of the program.

Reply:
Agreed. I added it while refactoring and forgot to remove it.